.github/workflows/check-docs-pr-preview.yml

```diff
@@ -0,0 +1,62 @@
+# This workflow checks that PRs modifying website docs include a preview link in the PR description.
+# It enforces KubeStellar's documentation PR best practices.
+
+name: Check Docs PR Preview Link
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+    paths:
+      - 'docs/content/**'
+      - 'docs/README.md'
+      - 'docs/mkdocs.yml'
+      - 'docs/overrides/**'
+      - 'docs/scripts/**'
+      - 'docs/requirements.txt'
+      - 'docs/main.py'
+
+jobs:
+  check-preview-link:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - name: Check for Preview Link in PR Description (bash)
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          REPO: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+        run: |
+          set -e
+          BODY=$(gh pr view "$PR_NUMBER" --repo "$REPO" --json body -q '.body')
+          DOC_BRANCH_REGEX='^doc-'
+          GITHUB_PAGES_URL="https://$(echo "$REPO" | cut -d'/' -f1).github.io/$(echo "$REPO" | cut -d'/' -f2)/$BRANCH"
+          PREVIEW_LINE=$(echo "$BODY" | grep -iE '^\s*Preview.*https?://')
+          GREEN='\033[0;32m'
+          RED='\033[0;31m'
+          YELLOW='\033[1;33m'
+          NC='\033[0m' # No Color
+          if [ -n "$PREVIEW_LINE" ]; then
+            # Extract the URL from the preview line (first URL found)
+            PREVIEW_URL=$(echo "$PREVIEW_LINE" | grep -oE 'https?://[^ ]*')
+            if [[ "$BRANCH" =~ $DOC_BRANCH_REGEX ]] && [[ "$PREVIEW_LINE" == *"$GITHUB_PAGES_URL"* ]]; then
+              # Check if the preview URL is fetchable (HTTP 200)
+              STATUS=$(curl -s -o /dev/null -w "%{http_code}" "$PREVIEW_URL")
+              if [ "$STATUS" = "200" ]; then
+                echo -e "${GREEN}[OK] Docs PR: doc- branch, valid preview link, and URL is fetchable.${NC}"
+              else
+                echo -e "${YELLOW}[WARN] Docs PR: Preview link present but URL is not fetchable (HTTP $STATUS).${NC}"
+              fi
+            elif [[ ! "$BRANCH" =~ $DOC_BRANCH_REGEX ]]; then
+              echo -e "${YELLOW}[WARN] Docs PR: Preview link found, but branch does not start with doc-. Please consider using a doc- branch for automatic previews.${NC}"
+            else
+              echo -e "${YELLOW}[WARN] Docs PR: Preview link found, but it may not point to the expected GitHub Pages URL.${NC}"
+            fi
+          else
+            echo -e "${RED}[ERROR] Docs PR: No preview link found. Please include a line starting with 'Preview' and a valid GitHub Pages URL for your changes.${NC}"
+            echo  # Ensure the last output always ends with a newline
+            echo
+            exit 1
+          fi
+          echo  # Ensure the last output always ends with a newline
```

Comment:
Do you see the red circle-dash here? This is the last thing I see in GitHub's display of this file. The red-circle-dash means that this file itself (NOT the output results from running this file) does NOT end in a newline. That is what I have been trying to talk about when I mentioned the lack of a terminal newline. That's "terminal" in the sense of "end", not "terminal" in the sense of "TTY output". I am NOT asking for the output from this file to end with a blank line. This is a minor annoyance and not a blocker.

<img width="171" alt="image" src="https://github.com/user-attachments/assets/b4886db2-7199-42ae-8aa7-73dc264c10bc" />